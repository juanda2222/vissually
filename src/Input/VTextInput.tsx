import React, { useContext } from 'react'


import styles from './VTextInput.module.css'
import styled, { ThemeContext } from 'styled-components';
import { Color2Vec } from "../Tools/ColorTools"

const VTextInput = ({
  className,
  theme,
  value,
  id,
  type,
  onChange,
}: InputProps) => {  

  const wrapped_className = [styles.input_text, className].join(" ")

  // create the theme depending 
  const contex_theme = useContext(ThemeContext);
  var input_theme;
  // use theme if a style is given
  if (!(typeof (contex_theme) == "undefined")) {
    input_theme = contex_theme
  } else {
    input_theme = theme
  }
  
  const rgb_list = Color2Vec(input_theme.color2)
  const StyledInput = styled.input`
    padding: 0.5em;
    margin: 0.5em;
    color: ${input_theme.color1};
    background: rgba(${rgb_list[0]}, ${rgb_list[1]}, ${rgb_list[2]}, 0.2);
    border: none;
    border-radius: 3px;
  `;
  
  return (
    <StyledInput
      type={type}
      className={wrapped_className} 
      id={id}
      value={value} 
      onChange={onChange}
    />
  )
}



VTextInput.defaultProps = {
  className: "",
  //type="text",
  //value="",
  theme: {
    textColor1: "#ffffff",
    textColor2: "#ffffff",
    color1: "#135b75",
    color2: "#135b75",
    backgroundColor1: "#ffffff",
    backgroundColor2: "#ffffff",
  }

};


export default VTextInput