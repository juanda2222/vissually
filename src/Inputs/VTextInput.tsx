import React, { useContext } from 'react'


import styles from './VTextInput.module.css'
import styled from 'styled-components';
import { Color2Vec } from "../Tools/ColorTools"
import { DefaultThemes, ThemeContext } from "../ThemeProvider/ThemeProvider"

const VTextInput: React.FunctionComponent<InputProps> = ({
  //styling properties
  className,
  style,
  theme,
  primary, secundary, dark,

  //functional properties
  value,
  id,
  type,
  onChange,
}) => {  

  // get the theme from the context
  const contex_theme:Theme = useContext(ThemeContext);
  const wrapped_className = [styles.input_text, className ? className : ""].join(" ")

  // generat the theme depending on the boolean inputs or the theme input
  var current_theme;
  if (secundary) {
    current_theme = DefaultThemes.secundary
  } else if (dark) {
    current_theme = DefaultThemes.dark
  } else if (primary) {
    current_theme = DefaultThemes.primary
  } else if (!(typeof (contex_theme) == "undefined")) {
    current_theme = contex_theme
  } else {
    current_theme = theme ? theme : DefaultThemes.primary
  }
  
  // generate the component from the style
  const rgb_list = Color2Vec(current_theme.color1)
  const StyledInput = styled.input`
    padding: 0.5em;
    margin: 0.5em;
    color: ${current_theme.textColor1};
    background: rgba(${rgb_list[0]}, ${rgb_list[1]}, ${rgb_list[2]}, 0.3);
    border: none;
    border-radius: 3px;
  `;
  
  return (
    <StyledInput
      type={type ? type : "text"}
      value={value} 
      className={wrapped_className} 
      style={style}
      id={id}
      onChange={onChange}
    />
  )
}


export default VTextInput