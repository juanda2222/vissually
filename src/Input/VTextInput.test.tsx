
import React from "react"
import renderer from 'react-test-renderer';
import VTextInput from './VTextInput'

describe('VButton', () => {

  test('Button declaration has no errors', () => {
    expect(VTextInput).toBeTruthy()
  })

  test("Default render. Snapshot test", () => { 
    const ButtonTree = renderer.create(<VTextInput/>).toJSON();
    console.log(ButtonTree)
    expect(ButtonTree).toMatchSnapshot();
  })

  test("Rendering with props. Snapshot test", () => { 
3

    const ButtonWithProps = renderer.create(
        <VTextInput
        className="some_classname"
        />
    ).toJSON();

    console.log(ButtonWithProps)
    expect(ButtonWithProps).toMatchSnapshot();
  })


})
