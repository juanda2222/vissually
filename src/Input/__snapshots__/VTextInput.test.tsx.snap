// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VButton Default render. Snapshot test 1`] = `
<input
  className="input_text"
  type="text"
/>
`;

exports[`VButton Rendering with props. Snapshot test 1`] = `
<input
  className="input_text"
  type="text"
/>
`;
